linode/nodepool/framework_models.go

```diff
@@ -0,0 +1,154 @@
+package nodepool
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+type NodePoolModel struct {
+	ID         types.Int64               `tfsdk:"id"`
+	ClusterID  types.Int64               `tfsdk:"cluster_id"`
+	Count      types.Int64               `tfsdk:"node_count"`
+	Type       types.String              `tfsdk:"type"`
+	Tags       types.Set                 `tfsdk:"tags"`
+	Nodes      types.List                `tfsdk:"nodes"`
+	Autoscaler []NodePoolAutoscalerModel `tfsdk:"autoscaler"`
+}
+
+type NodePoolAutoscalerModel struct {
+	Min types.Int64 `tfsdk:"min"`
+	Max types.Int64 `tfsdk:"max"`
+}
+
+type NodePoolNodeModel struct {
+	ID         types.String `tfsdk:"id"`
+	InstanceID types.Int64  `tfsdk:"instance_id"`
+	Status     types.String `tfsdk:"status"`
+}
+
+func flattenLKENodePoolLinode(node *linodego.LKENodePoolLinode) (*basetypes.ObjectValue, diag.Diagnostics) {
+	result := make(map[string]attr.Value)
+
+	result["id"] = types.StringValue(node.ID)
+	result["instance_id"] = types.Int64Value(int64(node.InstanceID))
+	result["status"] = types.StringValue(string(node.Status))
+
+	obj, errors := types.ObjectValue(nodeObjectType.AttrTypes, result)
+	if errors.HasError() {
+		return nil, errors
+	}
+	return &obj, nil
+}
+
+func parseNodeList(nodes []linodego.LKENodePoolLinode,
+) (*basetypes.ListValue, diag.Diagnostics) {
+	resultList := make([]attr.Value, len(nodes))
+	for i, node := range nodes {
+		result, errors := flattenLKENodePoolLinode(&node)
+		if errors.HasError() {
+			return nil, errors
+		}
+
+		resultList[i] = result
+	}
+	result, errors := basetypes.NewListValue(
+		nodeObjectType,
+		resultList,
+	)
+	if errors.HasError() {
+		return nil, errors
+	}
+
+	return &result, nil
+}
+
+func (pool *NodePoolModel) ParseNodePool(ctx context.Context, clusterID int, p *linodego.LKENodePool, diags *diag.Diagnostics) {
+	pool.ID = types.Int64Value(int64(p.ID))
+	pool.ClusterID = types.Int64Value(int64(clusterID))
+	pool.Count = types.Int64Value(int64(p.Count))
+	pool.Type = types.StringValue(p.Type)
+	tags, d := types.SetValueFrom(ctx, types.StringType, p.Tags)
+	if d != nil {
+		diags.Append(d...)
+		return
+	}
+	pool.Tags = tags
+
+	if p.Autoscaler.Enabled {
+		pool.Autoscaler = make([]NodePoolAutoscalerModel, 1)
+		pool.Autoscaler[0].Min = types.Int64Value(int64(p.Autoscaler.Min))
+		pool.Autoscaler[0].Max = types.Int64Value(int64(p.Autoscaler.Max))
```

Comment:
Readability nitpick:
```suggestion
		pool.Autoscaler = []NodePoolAutoscalerModel{
			{
				Min: types.Int64Value(int64(p.Autoscaler.Min)),
				Max: types.Int64Value(int64(p.Autoscaler.Max)),
			},
		}
```